bfd/elfxx-riscv.c

```diff
@@ -1760,6 +1761,66 @@ riscv_parse_add_implicit_subsets (riscv_parse_subset_t *rps)
     riscv_parse_add_subset (rps, "zicsr",
 			    RISCV_UNKNOWN_VERSION,
 			    RISCV_UNKNOWN_VERSION, TRUE);
+  else if ((riscv_lookup_subset (rps->subset_list, "k", &subset)))
```

Comment:
I suppose k, zkn and zks are similar to g, they used to expand into others.  We won't output g to the final architecture string, so k, zkn and zks should be the same.  I had done some improvement on FSF binutils about this, so it might be easier to complete over there.

Reply:
However,my old implementation here did not consider so much, so it looks OK.

Reply:
Okay, if it need change, I will update it.